### PR TITLE
docs(runtime): explicit dispatch example + JITFunction.compile() guide

### PR DESCRIPTION
**Pull request by lyfne123**
## Summary

Pure docs + example PR. Surface the runtime APIs that landed in PR #1496 (CompiledProgram extraction), PR #1599 (Worker ABC + ChipWorker / DistributedWorker + run/register/RegistrationHandle), and PR #1620 (JITFunction.compile()) so external users have a single place to learn the new patterns.

No code changes.

### Example: `examples/runtime/explicit_dispatch.py`

Two @pl.jit kernels + one shared ChipWorker, exercising three end-to-end patterns:

- **mode_a_inference_service** — pre-register N compiled kernels, dispatch by name (the serving-runtime pattern from issue #1455).
- **mode_b_training_loop** — persistent weight DeviceTensor via `alloc_tensor(init=host_weight)` plus a hot register/run loop with per-step inputs.
- **mode_c_register_dispatch_overhead** — assert `aicpu_dlopen_count` stays at +1 across 20 dispatches after a single `register()`.

All three modes share a single `ChipWorker` (one init / one close) and use `JITFunction.compile()` for stage-1 compile + `worker.register()` for stage-2 dispatch.

### Docs: `docs/en/user/00-getting_started.md` + ZH

New "**Explicit dispatch (`worker.run`, `worker.register`)**" subsection between the existing DeviceTensor block and the distributed block. Covers:

- When to prefer explicit dispatch over implicit `with ChipWorker(): compiled(...)`.
- `RegistrationHandle` semantics — cid alias-safety, `with handle:` lifetime, `unregister()` early-release.
- The same flow on a @pl.jit kernel via `JITFunction.compile()`.
- Link to the new example.

### Docs: `docs/en/user/01-language_guide.md` + ZH

New "**`JITFunction.compile()` (for `@pl.jit` kernels)**" subsection under the existing Compilation section. Documents the compile-only entry point added in PR #1620:

- Splits compile / runtime stages for @pl.jit kernels.
- Same `config=RunConfig(...)` forwarding semantics as `__call__`.
- Cache-shared with `__call__` — no double compile.
- Returns a `CompiledProgram` exposing the full extraction surface so harnesses can drive `simpler.worker.Worker` directly without writing a `@pl.program` wrapper.

## Related

- Issue #1455 — closed by PR #1620 (Worker.run / register integration for @pl.jit).
- Issue #1404 — also closed by PR #1620 (duplicate of #1455 — separate compile/runtime stages).
- PR #1496 — CompiledProgram extraction surface (chip_callable / build_orch_args / build_call_config).
- PR #1599 — Worker ABC + ChipWorker / DistributedWorker hierarchy.
- PR #1620 — JITFunction.compile() — the entry point this PR demonstrates.

## Testing

- [x] Example file lints (ruff + pyright via pre-commit).
- [x] Markdown lints via pre-commit.
- [ ] On-device run of `python examples/runtime/explicit_dispatch.py` to confirm all three modes hit their `assert` checks. Recommended as a smoke test before merging.

The example uses fully static shapes and stock fp32 add/mul kernels, so it should be the smallest possible reproducer of the new APIs against a real device.

### CODE REVIEW

File: examples/runtime/explicit_dispatch.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+    assert after_register == before + 1, (
+        f"register should dlopen exactly once: before={before}, after_register={after_register}"
+    )
+    assert after_runs == after_register, (
+        f"20 dispatches should not re-dlopen: after_register={after_register}, after_runs={after_runs}"
+    )
+    print(
+        f"mode C OK — aicpu_dlopen_count: {before} -> {after_register} (1 register) -> {after_runs} (20 runs)"
+    )
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply examples/runtime/explicit_dispatch.py]: Fixed in `f293c5eb` — mode C now instantiates its own ChipWorker so the first `register(tile_add_128)` is observable. Shared-worker mode A had already populated `_cid_cache` and would have masked the dlopen bump. The script now uses two workers: shared(A+B) + isolated(C).

File: examples/runtime/explicit_dispatch.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    # The weight DeviceTensor has ``child_memory=True`` semantics — no H2D
+    # cost per step.
+    for step in range(5):
+        x = torch.full((128, 128), float(step), dtype=torch.float32)
+        out = torch.zeros((128, 128), dtype=torch.float32)
+        h(x, w_dev, out)
+        expected = x + host_weight
+        assert torch.allclose(out, expected, rtol=1e-5, atol=1e-5), f"step {step} mismatch"
+
+    print("mode B OK — 5 dispatches with persistent weight DeviceTensor")
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply examples/runtime/explicit_dispatch.py:136]: Fixed in `f293c5eb` — added explicit `worker.free_tensor(w_dev)` after the loop with a comment noting close() would auto-free but explicit is the documented best practice.

File: examples/runtime/explicit_dispatch.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+# ---------------------------------------------------------------------------
+# Kernels (shared across the three modes)
+# ---------------------------------------------------------------------------
+
+
+@pl.jit
+def tile_add_128(a: pl.Tensor, b: pl.Tensor, c: pl.Out[pl.Tensor]):
+    """c = a + b on 128x128 fp32 tiles."""
+    with pl.incore():
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply examples/runtime/explicit_dispatch.py]: Fixed in `f293c5eb` — replaced both `with pl.incore():` instances with `with pl.at(level=pl.Level.CORE_GROUP):` per the language guide's deprecation note.

File: examples/runtime/explicit_dispatch.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        tile_b = pl.load(b, [0, 0], [128, 128])
+        tile_c = pl.add(tile_a, tile_b)
+        pl.store(tile_c, [0, 0], c)
+    return c
+
+
+@pl.jit
+def tile_mul_128(a: pl.Tensor, b: pl.Tensor, c: pl.Out[pl.Tensor]):
+    """c = a * b on 128x128 fp32 tiles."""
+    with pl.incore():
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply examples/runtime/explicit_dispatch.py]: Fixed in `f293c5eb` — same change as the mode A kernel; both kernels now use `pl.at(level=pl.Level.CORE_GROUP)`.

File: examples/runtime/explicit_dispatch.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+# ---------------------------------------------------------------------------
+# Kernels (shared across the three modes)
+# ---------------------------------------------------------------------------
+
+
+@pl.jit
+def tile_add_128(a: pl.Tensor, b: pl.Tensor, c: pl.Out[pl.Tensor]):
+    """c = a + b on 128x128 fp32 tiles."""
+    with pl.incore():
+        tile_a = pl.load(a, [0, 0], [128, 128])
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply examples/runtime/explicit_dispatch.py]: Fixed in `f293c5eb` — both `pl.incore()` usages replaced with `pl.at(level=pl.Level.CORE_GROUP)` (same change gemini flagged on lines 45 and 56).

File: docs/en/user/01-language_guide.md [hunk 6]
```diff
@@ -585,6 +585,46 @@ The `Default` strategy runs these passes in order:
...
+print("artifacts in:", compiled.output_dir)
+
+# Stage 2: explicit runtime via the new worker API.
+from pypto.runtime import ChipWorker, RunConfig
+
+worker = ChipWorker(config=RunConfig(platform="a2a3sim"))
+handle = worker.register(compiled)
+for batch in stream:
+    handle(batch.x, w_dev, batch.out)
+worker.close()
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/01-language_guide.md:614]: Fixed in `f293c5eb` — Stage-2 snippet now allocates `w_dev` via `worker.alloc_tensor(sample_w.shape, sample_w.dtype, init=sample_w)` and releases with `worker.free_tensor(w_dev)` before `worker.close()`. Variable lifecycle is now end-to-end traceable.

File: docs/zh-cn/user/01-language_guide.md [hunk 7]
```diff
@@ -571,6 +571,44 @@ output_dir = ir.compile(
...
+print("artifacts in:", compiled.output_dir)
+
+# Stage 2：用新的 worker API 显式驱动 runtime
+from pypto.runtime import ChipWorker, RunConfig
+
+worker = ChipWorker(config=RunConfig(platform="a2a3sim"))
+handle = worker.register(compiled)
+for batch in stream:
+    handle(batch.x, w_dev, batch.out)
+worker.close()
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/zh-cn/user/01-language_guide.md:599]: Fixed in `f293c5eb` — 中文文档同步更新,Stage-2 代码块加上了 `alloc_tensor` + `free_tensor`,与英文文档保持一致。

**Review by Hzfengsy — APPROVED** [review]
